### DISCUSSION TITLE
Clarification on data format

### DISCUSSION THREAD

**Discussion by cj2001** [snap-stanford/ogb#171] [WikiKG90M-LSC]
I have a quick question on the data format.  The docs show:

```
train_hrt = dataset.train_hrt # numpy ndarray of shape (num_triplets, 3)
i = 0
print(train_hrt[i]) 
```

The docs say "Each triplet (head, relation, tail) in WikiKG90M-LSC represents an Wikidata claim, where head and tail are the Wikidata items, and relation is the Wikidata predicate."  So I took away from that that these values are Wikidata identifiers (example `Q42` is Douglas Adams).  However, if I look at what the format of these triples are `print(train_hrs[0:10])` what I get is

```
[[      0    1182  650146]
 [      0    1182  650332]
 [      0     876  989978]
 [      0     168 1054521]
 [      0    1182 1632931]
 [      0     876 2437486]
 [      0     324 2705654]
 [      0     324 5885915]
 [      0     427 6972024]
 [      0     427 7156732]]
```

If I take the head of that first triples, I had assumed this might be something like `Q0`, however `Q0` does not exist in Wikidata.

So is there a way to map these integers to the text in Wikidata?  

Thanks in advance!

**Comment by hyren** [comment]
Hi, the numbers in train_hrt are NOT Wikidata identifiers, but rather the node id/relation id in the KG. For example, the first row [0 1182 650146] represents there is the relation 1182 between node 0 and node 650146. And these idx ranges from [0, num_node) and [0, num_relation) on the KG.

We currently do not provide the raw text, but will probably release them in the future (as in https://github.com/snap-stanford/ogb/discussions/147#discussioncomment-601875). Thank you.

**Reply by cj2001** [reply]
Cool.  I replied in that thread throwing in my support for you all eventually (hopefully soon!) putting that raw text data out there.  Cheers!
